src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -493,17 +494,10 @@ class MacroAssembler: public Assembler {
 
   inline void mov(Register dst, address addr)             { mov_immediate64(dst, (uint64_t)addr); }
 
-  inline void mov(Register dst, int imm64)                { mov_immediate64(dst, (uint64_t)imm64); }
-  inline void mov(Register dst, long imm64)               { mov_immediate64(dst, (uint64_t)imm64); }
-  inline void mov(Register dst, long long imm64)          { mov_immediate64(dst, (uint64_t)imm64); }
-  inline void mov(Register dst, unsigned int imm64)       { mov_immediate64(dst, (uint64_t)imm64); }
-  inline void mov(Register dst, unsigned long imm64)      { mov_immediate64(dst, (uint64_t)imm64); }
-  inline void mov(Register dst, unsigned long long imm64) { mov_immediate64(dst, (uint64_t)imm64); }
+  template<typename T, ENABLE_IF(std::is_integral<T>::value)>
+  inline void mov(Register dst, T o)                      { mov_immediate64(dst, (uint64_t)o); }
```

Comment:
I'm not sure why this isn't just
`inline void mov(Register dst, uint64_t o) { mov_immediate64(dst, o); }`
Unlike the `Address` case, I don't see any ambiguous implicit conversions here.

Reply:
Because the macosx compiler complains of ambiguity for the mov also if this is the only mov.